api/cds.proto

```diff
@@ -417,6 +415,15 @@ message Cluster {
   // For instance, if the metadata is intended for the Router filter, the filter
   // name should be specified as *envoy.router*.
   Metadata metadata = 25;
+
+  enum ClusterProtocolSelection {
+    // Cluster can only operate on one of the possible upstream protocols (HTTP1.1, HTTP2).
+    // If http2_protocol_options are present, HTTP2 will be used, otherwise HTTP1.1 will be used.
```

Comment:
Are there any load balancing implications when using mixed upstream protocols?

Reply:
Current Envoy implementation assumes pooled connections are all either HTTP/1 or HTTP/2, as open connections are reused without any version checks. My implementation expands the containers currently used for connection pools to factor in the protocol version into the pool selection.

Reply:
In the default mode the cluster there is no change, so no impact.
In the 'USE_DOWNSTREAM_PROTOCOL' each request is load balanced on the set of hosts as before, and the connection pooling extension I mentioned above makes sure that v1&2 connections are not mixed up. With this I don't see any load balancing impact.